rasa/train.py

```diff
@@ -174,7 +174,7 @@ async def _train_async_internal(
             additional_arguments=nlu_additional_arguments,
         )
 
-    if nlu_data.is_empty():
+    if nlu_data.can_train_nlu_model():
         print_warning("No NLU data present. Just a Rasa Core model will be trained.")
```

Comment:
will it train nlu featurizers?

Reply:
good point, no

We have a test that checks that if only a core model was trained, by default a RegexInterpreter will be used during inference time. If this would be `nlu_data.is_empty` and we have core data but no nlu data, it will train a KeywordClassifier by default, which restulted the test to fail.

Reply:
but that might be something that we can tackle in the `e2e` branch as we don't have proper `e2e` yet?

Reply:
Ah, just took a closer look at the test, it actually uses a config with a `KeywordIntentClassifier`, so that is why the classifier was trained. So, I guess we could also update the test `test_test` to use a empty pipeline? Or would we use the default pipeline then? Also, in what kind of situations should we not train NLU when the user executed `rasa train`?

Reply:
>in what kind of situations should we not train NLU when the user executed rasa train?

if there is no data at all or no nlu config

> it actually uses a config with a KeywordIntentClassifier,

why a classifier will be trained if there is no data?

we should train nlu (the components that are in nlu config) if there is data from core (but only those components that have data).

Reply:
I guess we can tackle it in e2e or minor

Reply:
> if there is data from core

This exclude data from default actions, right?

Reply:
> why a classifier will be trained if there is no data?

Good question. It is basically an empty `KeywordIntentClassifier`. 

Yeahc let's tackle this in e2e and make sure to cover all possible scenarios.

Reply:
we can include them, they exist if core data exists

Reply:
@tabergma could you please add it to the list in e2e for real issue?